Add fixes to support game version 1.5.9

MaxBosse opened the pull request:
## PR Checklist
<!-- Insert "x" inside square brackets to check item. -->

Please check if your PR fulfills the following requirements:

- [ ] All new classes have class-level documentation comments, if there are any at all
- [ ] Tests for the changes have been added (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?
<!-- Please check the one that applies to this PR. -->

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactoring (no functional changes, no api changes)
- [x] Build related changes
- [ ] CI related changes
- [ ] Documentation update
- [ ] Other... Please describe:


## What is the current behaviour?
Currently you are unable to build the project with the game version 1.5.9


## What is the new behaviour?
This PR fixes all errors during the build of the project and allows you to start the game with the mod and host a coop campaign.


## Other information
The OnLordPartySpawned event was removed in the last update. To compensate I wrote a patch that executes a custom event after a call to InitializeLordPartyProperties.
In 1.5.8 the event was called at the end of InitializeLordPartyProperties, which is why I decided to use a suffix instead of the usualy prefix.

MaxBosse committed:
[FIX] Removal of OnLordPartySpawnedEvent

MaxBosse committed:
[FIX] Use new IGauntletMovie type

MaxBosse committed:
[FIX] Adjust to new namespaces for CharacterCreation

MaxBosse committed:
[FIX] Overwrite ExecuteSelection function

MaxBosse committed:
[WIP] Comment out NotImplementedExceptions

Currently these exceptions will be thrown once you reach the review
stage of the character creations. To proceed I commented them out
for now.

MaxBosse committed:
[WIP] Fix initialization of the LordPartyComponent

The clan parameter was removed from the method.
It is now received via the MobileParty, which at that point is not set
and causing the game to error out.

System.NullReferenceException
  HResult=0x80004003
  Message=Object reference not set to an instance of an object.
  Source=TaleWorlds.CampaignSystem
  StackTrace:
   at TaleWorlds.CampaignSystem.WarPartyComponent.OnFinalize()
   at TaleWorlds.CampaignSystem.MobileParty.set_PartyComponent(PartyComponent value)
   at Coop.Mod.Serializers.PlayerHeroSerializer.Deserialize() in G:\Project\BannerlordCoop\source\Coop\Mod\Serializers\CustomSerializers\PlayerHeroSerializer.cs:line 181

p-JordanStiebritz commented on source/Coop/Mod/Persistence/MobilePartyEntityManager.cs:
```diff
@@ -116,7 +116,7 @@ private void InitRoom(RailServerRoom room)
             
             CampaignEvents.MobilePartyCreated.AddNonSerializedListener(this, OnPartyAdded);
-            CampaignEvents.OnLordPartySpawnedEvent.AddNonSerializedListener(this, OnPartyAdded);
+            LordPartyComponentPatch.OnLordPartySpawned += (sender, e) => OnPartyAdded(e);
```
Might be able to just get away with removing this line wholly rather than adding `LordPartyComponentPatch`. Since 1.5.9 changed it so that lord parties are just mobile parties, we would probably be handling the party being added twice for players. (Might be a reason why the main party spawned and AI took control of it?)

MaxBosse replied:
Removing the patch indeed fixed the duplicate party. Thanks!

MaxBosse committed:
[FIX] Remove LordPartyComponentPatch

Lord parties are normal mobile parties in 1.5.9 so this is not needed.
Including it actually causes a bug resulting in duplicate lord-parties.

MaxBosse committed:
[FIX] Set the ActualClan based on the PlayerClan

MaxBosse committed:
[CLEANUP] Remove commented out line

p-JordanStiebritz commented:
Looks fine to me. I think there is a previous existing issue where when you exit, it doesn't actually despawn you from the game. This could be fine, but requires us to not register the client's party a second time or to correctly unregister it when player exits or the game crashes. Any opinions on the persistent client party @garrettluskey ?
